### ISSUE TITLE
didOpenExternalApplication not reliably called

### ISSUE THREAD

**Issue by leslieolivier** [Adyen/adyen-ios#779]
When users are indirectly app switched to an external app (via `BrowserComponent`), sometimes didFail(ComponentError.cancelled) is called instead of the correct `didOpenExternalApplication()`. This results in users authorising a payment In their external app for an order which has already been cancelled in our system.

For an external app like Vipps, we seem to have around 15% of payment attempts where this happens.

I think the problem could be in:

```
    /// This allows us to assume one of the following scenarios:
    /// - SFSafariViewController deliberately closed by user and current app still in foreground;
    /// - SFSafariViewController finished due to a successful redirect to an external app and current app no longer in foreground.
    private func finish() {
        DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(25)) {
            if UIApplication.shared.applicationState == .active {
                self.delegate?.didCancel()
            } else {
                self.delegate?.didOpenExternalApplication()
            }
        }
    }
```

We have recently upgraded from Adyen Components v3 to v4. When using v3 we had implemented a similar method, but we waited 1 second before deciding whether to call `didCancel` or `didOpenExternalApplication` and we did not have the same amount of wrong `didCancel`s when using 1 second as delay.

So I think, it might be necessary to increase the delay in the SDK to minimise the wrong `didCancel()` calls.

We are currently using iOS SDK 4.4.0.

Thanks in advance!

**Comment by mohammedDehairy** [comment]
Hi @leslieolivier ,

Thanks so much for the feedback!.

indeed 25 milliseconds might be a too small delay, so we're gonna fix it in the next version of v4.

Mohamed

**Comment by descorp** [comment]
Should be fixed in this PR #780

**Comment by descorp** [comment]
Released in [4.7.0](https://github.com/Adyen/adyen-ios/releases/tag/4.7.0)

@leslieolivier could you verify?

**Comment by leslieolivier** [comment]
Hi!

I have not yet released our app with the 4.7.0 SDK, but I would assume it will, because it is exactly what I used in v3 of the SDK.

**Comment by descorp** [comment]
Then I will close this ticket for now. Feel free to continue here if you have any additional feedback.
